pkg/util/file_stats.go

```diff
@@ -0,0 +1,65 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package util
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/shirou/gopsutil/v3/process"
+)
+
+type FileStats struct {
+	AgentOpenFiles float64 `json:"agent_open_files"`
+	OsFileLimit    float64 `json:"os_file_limit"`
+}
+
+// getFileStats returns the number of files the Agent process has open
+func GetFileStats() *FileStats {
+	stats := FileStats{}
+
+	// Creating a new process.Process type based on Agent PID
+	p, err := process.NewProcess(int32(os.Getpid()))
+	if err != nil {
+		log.Errorf("Failed to retrieve agent process: %s", err)
+		return &stats
+	}
+
+	// Retrieving []OpenFilesStat from Agent process.Process type
```

Comment:
I believe we can consolidate a few of these calls into gopsutil into just a single call to `p.RlimitUsage(true)`. According to the [docs](https://pkg.go.dev/github.com/shirou/gopsutil@v3.21.11+incompatible/process#Process.RlimitUsage), this single call will fill in both the limits and the current usage of said resource. 

So it seems like you can just do that as a single call and then you'll have access to both the current usage and the limit.